### PR TITLE
Ansible:Nagios:Server - Automate Base Server Installation

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes issue #2609 

Implements task 1 of the Nagios refresh epic ( Issue #2445 )

N.B The new nagios playbook, installs only a base server, from which the existing live configuration can be applied, or ( once later tasks are completed on the epic ) generated from source.

The readme.md [here](https://github.com/steelhead31/infrastructure/blob/fix_2609/ansible/playbooks/nagios/README.md) has some more detailed technical information.

Prior to submitting this PR, the following items need to be completed :

- [x] Find appropriate place to store encrypted secrets file passwod
- [x] Obtain new nagios/slack webhook URL
- [x] Update encrypted secrets file with new nagios admin password and webhook

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [X] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [X] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

File: ansible/playbooks/nagios/roles/Nagios_Server/scripts/slack_nagios.pl [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/scripts/slack_nagios.pl:1]: rename the file to slack_nagios.sh ?

File: ansible/playbooks/nagios/roles/Nagios_Server/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+# defaults file for Nagios_Installation
+#
+
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/defaults/main.yml:4]: add "admin_user": nagiosadmin as variable ?

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    port: 80/tcp
+    permanent: yes
+    state: enabled
+  become: yes
+  when: ansible_distribution == "RedHat" or ansible_distribution == "CentOS"
+
+- name: Create nagiosadmin user for Nagios Instance
+  htpasswd:
+    path: /usr/local/nagios/etc/htpasswd.users
+    name: nagiosadmin
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml:94]: use variable admin_user than hardcoded "nagiosadmin"

File: ansible/playbooks/nagios/vars_setup_server.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (zdtsw) [inline ansible/playbooks/nagios/vars_setup_server.yml:1]: the playbook `play_setup_server.yml `includes this `vars_serup_server.ym`l and include role Nagios_Server which has a `defaults`.
these two files seem duplicated. 
do we really need this file?

File: ansible/playbooks/nagios/roles/Nagios_Server/scripts/commands.cfg [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+#define command{
+#	command_name	notify-service-by-email
+#	command_line	/usr/bin/printf "%b" "***** Nagios *****\n\nNotification Type: $NOTIFICATIONTYPE$\n\nService: $SERVICEDESC$\nHost: $HOSTALIAS$\nAddress: $HOSTADDRESS$\nState: $SERVICESTATE$\n\nDate/Time: $LONGDATETIME$\n\nAdditional Info:\n\n$SERVICEOUTPUT$\n" | /bin/mail -s "** $NOTIFICATIONTYPE$ Service Alert: $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **" $CONTACTEMAIL$
+#	}
+
+
+define command {
+	command_name notify-service-by-slack
+command_line /usr/local/nagios/bin/slack_nagios.pl -field slack_channel=#infrastructure-bot -field HOSTALIAS="$HOSTNAME$" -field SERVICEDESC="$SERVICEDESC$" -field SERVICESTATE="$SERVICESTATE$" -field SERVICEOUTPUT="$SERVICEOUTPUT$" -field NOTIFICATIONTYPE="$NOTIFICATIONTYPE$"
+# command_line /usr/local/bin/slack_nagios.pl -field slack_channel=#infrastructure-bot -field HOSTALIAS="$HOSTNAME$" -field SERVICEDESC="$SERVICEDESC$" -field SERVICESTATE="$SERVICESTATE$" -field SERVICEOUTPUT="$SERVICEOUTPUT$" -field NOTIFICATIONTYPE="$NOTIFICATIONTYPE$"
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/scripts/commands.cfg:42]: want to keep these commented command_line?

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+- name: Run Nagios-Core Installation
+  include_tasks: install_nagios_core.yml
+  when: '"install_core" in task_list'
+
+- name: Run Nagios-Plugins installation
+  include_tasks: install_nagios_plugins.yml
+  when: '"install_plugins" in task_list'
+
+- name: Run PNP$NAGIOS-Plugin installation
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/main.yml:15]:
```suggestion
- name: Run PNP4NAGIOS-Plugin installation
```

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+- name: Download + Unpack Nagios Core {{ nagios_version }} source
+  unarchive:
+    src: "https://assets.nagios.com/downloads/nagioscore/releases/nagios-{{ nagios_version }}.tar.gz"
+    remote_src: yes
+    dest: "/tmp"
+  when: not nagios_installation.stat.exists
+
+# 'Creates' argument keeps the playbook idempotent.
+- name: Run './configure' on source.
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml:33]:
```suggestion
- name: "Run '/tmp/nagios-{{ nagios_version }}/configure' on source."
```

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  command:
+    cmd: "./configure {{ extra_configure_args }}"
+    chdir: "/tmp/nagios-{{ nagios_version }}/"
+    creates: "/tmp/nagios-{{ nagios_version }}/Makefile"
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml:37]: do we need to have this pre-check for existence of Makefile? assume nagios_installation is not installed on this machine then does not really matter if the same version nagios already exists in /tmp

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    - rewrite
+    - cgi
+  loop_control:
+    loop_var: module
+  become: yes
+  when:
+    - '"install-webconf" in make_list'
+    - ansible_distribution == "Ubuntu"
+
+- name: Configure Firewalld for RHEL
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml:82]: not sure if we need this block, based on everything here is only for Ubuntu

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml (hunk 3 above)
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml:94]: good to have `nagiosadmin` as variable alongwith `nagios_admin_pass`

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagiosgraph.yml [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+- name: Update Use Lib Path In CGI Files
+  replace:
+    path: "{{ item.path }}"
+    regexp: "use lib '/opt/nagiosgraph/etc';"
+    replace: "use lib '/usr/local/nagios/nagiosgraph/etc';"
+  with_items: "{{ cgifiles.files }}"
+  become: yes
+  when: not nagiosgraph_plugins_installation.stat.exists
+
+- name: copy nagiosgraph cgi files into place
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagiosgraph.yml:51]:
```suggestion
- name: Copy nagiosgraph cgi files into place
```

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagiosgraph.yml [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+  copy:
+    src: "{{ item.path }}"
+    dest: "/usr/local/nagios/sbin"
+    remote_src: yes
+    # owner: nagios
+    # group: nagios
+    mode: 0775
+  with_items: "{{ cgifiles.files }}"
+
+- name: Update Use Lib Path In Insert.PL script
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagiosgraph.yml:61]:
```suggestion
- name: Update Use Lib Path In insert.pl Script
```

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagiosgraph.yml [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+- name: Copy The Slack NagiosGraph Perl Script Into Alt. Location
+  copy:
+    src: /usr/local/nagios/nagiosgraph/lib/insert.pl
+    remote_src: yes
+    dest: /usr/local/nagios/libexec/insert.pl
+    # owner: nagios
+    # group: nagios
+    mode: '0775'
+
+- name: copy Nagiosgraph CSS & JS files into place
```
Comment (zdtsw) [inline ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagiosgraph.yml:93]:
```suggestion
- name: Copy Nagiosgraph CSS & JS files into place
```

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml (hunk 9 above)
Comment (steelhead31) [reply ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml:81]: Good point! - Fixed

File: ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml (hunk 3 above)
Comment (steelhead31) [reply ansible/playbooks/nagios/roles/Nagios_Server/tasks/install_nagios_core.yml]: Yup,  Fixed that as well..

**Comment by steelhead31** [comment]
Remaining linter errors are not related to changes made in this PR.

**Review by sxa — APPROVED** [review]
LGTM on the basis that it's been tested. Will be good to have this committed and it can be refined later if desired.
Added Martijn as a review in case he wants to provide another set of eyeballs ...

**Review by zdtsw — APPROVED** [review]

File: ansible/playbooks/nagios/README.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+    ansible-vault edit secrets_setup_server.enc
+
+    The Encrypted File Can Have Its Password Changed With The Following command
+
+    ansible-vault rekey secrets_setup_server.enc
+
+
+**Usage Guide :**
+
```
[the remark this answers is not in the document]
Comment (karianna) [reply ansible/playbooks/nagios/README.md:55]: I think folks know to ask in infra for escalated privileges there

File: ansible/playbooks/nagios/roles/Nagios_Server/scripts/commands.cfg (hunk 5 above)
Comment (karianna) [reply ansible/playbooks/nagios/roles/Nagios_Server/scripts/commands.cfg:42]: ideally should remove commented out code

File: ansible/playbooks/nagios/roles/Nagios_Server/scripts/commands.cfg [hunk 14]
```diff
@@ -0,0 +1,377 @@
...
+         command_name  check_label
+         command_line  $USER1$/check_label "$ARG1$" $ARG2$ $ARG3$
+}
+
+# https checks with check_http
+define command{
+	command_name    check_https_url
+	command_line    $USER1$/check_http --ssl -H '$HOSTADDRESS$' -u $ARG1$
+}
+# Check for Updates using a key
```
Comment (karianna) [inline ansible/playbooks/nagios/roles/Nagios_Server/scripts/commands.cfg:348]: more commented out code?

File: ansible/playbooks/nagios/roles/Nagios_Server/scripts/slack_nagios.pl [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+elif [ "$NAGIOS_SERVICESTATE" = "UNKNOWN" ]
+then
+    ICON_EMOJI=":question:"
+else
+    ICON_EMOJI=":fire:"
+fi
+
+
+#request for posting to a channel
+curl -X POST --data "payload={\"channel\": \"${SLACK_CHANNEL}\", \"username\": \"${SLACK_BOTNAME}\", \"icon_emoji\": \":nagios:\", \"text\": \"${ICON_EMOJI} HOST: ${NAGIOS_HOSTNAME}   SERVICE: ${NAGIOS_SERVICEDISPLAYNAME} STATE: ${NAGIOS_SERVICESTATE} MESSAGE: ${NAGIOS_SERVICEOUTPUT} <https://${WEBHOST_NAGIOS}/nagios/cgi-bin/status.cgi?host=${NAGIOS_HOSTNAME}|See Nagios>\"}" ${WEBHOOK_URL}
```
[the remark this answers is not in the document]
Comment (karianna) [reply ansible/playbooks/nagios/roles/Nagios_Server/scripts/slack_nagios.pl:26]: I think that's to be expected?

**Review by karianna — CHANGES_REQUESTED** [review]
Some nitpick comments and a small amount of lint fixes to go

File: ansible/playbooks/nagios/roles/Nagios_Server/scripts/slack_nagios.pl (hunk 15 above)
[the remark this answers is not in the document]
Comment (zdtsw) [reply ansible/playbooks/nagios/roles/Nagios_Server/scripts/slack_nagios.pl:26]: yeah, I got user account created by @steelhead31 and it works now.

**Review by Haroon-Khel — APPROVED** [review]

**Comment by sxa** [comment]
> a small amount of lint fixes to go

Are you referring to these ones?
```
Error: name Jinja templates should only be at the end of 'name'
```
I think all of the things in this PR correctly adhere to that rule.

Having said that, I'm not convinced I agree with [the policy enforced by the rule](https://ansible-lint.readthedocs.io/rules/name/) and I'm tempted to suggest removing it. We've got some cases where it doesn't make sense (e.g. in path names).

**Comment by sxa** [comment]
> Having said that, I'm not convinced I agree with [the policy enforced by the rule](https://ansible-lint.readthedocs.io/rules/name/) and I'm tempted to suggest removing it.

PR to skip it: https://github.com/adoptium/infrastructure/pull/2775

**Review by karianna — APPROVED** [review]

**Pull request merged**